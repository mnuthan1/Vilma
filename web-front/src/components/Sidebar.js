import React, { useState } from 'react';

import {logout} from '../services/user.service'

/**
 * Main page Sidebar
 * @param {object} props Component props
 * @visibleName sidebar
 */
export function Sidebar({ props }) {
  return (
    <div className="vertical-nav bg-white" id="sidebar">
      <LogoHeader />
      <Profile profile={props.profile} />
      <MenuList menuList={props.menuList} />
    </div>
  );
}

/**
 * Logo Header
 * @param {object} props Component props
 * @visibleName Sidebar header with logo
 */
export function LogoHeader() {
  return (
    <nav className="navbar navbar-light bg-info">
      <a className="navbar-brand" href="#">
        <span className="button"> 
          <img width="30" className="mr-3"  src = "https://www.designevo.com/res/templates/thumb_small/green-and-blue-circle.png"/>
        </span>
        Navbar
      </a>
    </nav>
  )
}
/**
 * User profile details
 * @param {object} props Component props
 * @param {object} props.profile user details
 * @visibleName user details
 */
export function Profile({ profile }) {
  return (
    <div className="py-1 px-3 mb-3 bg-light">
      <div className="media d-flex align-items-center">
        <img
          src={profile.avatar} alt="..."
          width="65" className="mr-3 rounded-circle img-thumbnail shadow-sm"
        />
        <div className="media-body">
          <h4 className="m-0">{profile.name}</h4>
          <p className="font-weight-light text-muted mb-0">{profile.designation}</p>
        </div>
      </div>
      <Logout />
    </div>
  )
}

/**
 * Logout
 * @param {object} props Component props
 * @visibleName Logout
 */

export function Logout() {
  return (
    <div className="w-100 row .text-gray" >
      <i className="fa mr-3 fa-fw fa-power-off" onClick={logout}></i>
    </div>
  )
}
/**
 * MenuList with list of menus
 * @param {object} props Component props
 * @param {object} props.menuList single menu details
 * @visibleName MenuList with list of menus
 */
export function MenuList({ menuList }) {
  return (
    menuList.map( (menu, i) => {
      return <Menu menu={menu} key={i}/>;
    })
  )
}

/**
 * Menu with list of commands
 * @param {object} props Component props
 * @param {object} props.menu single menu details
 * @visibleName Menu with list of commands
 */
export function Menu({ menu }) {
  const commandList = menu.commands;
  const listItems = menu.commands.map((cmd, i) =>
    <Command cmd={cmd} key={i}/>
  );
  return (
    <React.Fragment>
      <p className="text-gray font-weight-bold text-uppercase px-3 small pb-4 mb-0">{menu.displayName}</p>
      <ul className="nav flex-column bg-white mb-0">
        {listItems}
      </ul>
      </React.Fragment>
  )
}

/**
 * Command for Menu item
 * @param {object} props Component props
 * @param {object} props.cmd single command details
 * @visibleName The Best Menu Item (Command) Ever üêô
 */
export function Command({ cmd }) {
  const className = "fa mr-3 text-primary fa-fw " + cmd.icon;
  return (
    <li className="nav-item">
      <a href={cmd.action} className="nav-link text-dark font-italic bg-light">
        <i className={className}></i>
        {cmd.displayName}
      </a>
    </li>
  )
}