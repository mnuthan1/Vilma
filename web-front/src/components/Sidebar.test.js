import React from 'react';
import { Command, Menu, MenuList, Sidebar } from './Sidebar';

import renderer from 'react-test-renderer';

/**
 * jtest case for react component Command
 */
describe('Command', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Command cmd={{"action":"#",
      "displayName":"Home",
      "icon": "fa-th-large"}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/**
 * jtest case for react component Menu
 */
describe('Menu', () => {
  const menu = {"displayName": "Main",
  "commands": [
    {"action":"#",
     "displayName":"Home",
     "icon": "fa-th-large"},
     {"action":"#",
     "displayName":"About",
     "icon": "fa-address-card"},
     {"action":"#",
     "displayName":"Services",
     "icon": "fa-cubes"},
     {"action":"#",
     "displayName":"Gallery",
     "icon": "fa-picture-o"}
  ]}

  it('renders correctly', () => {
    const tree = renderer
      .create(<Menu menu={menu} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


/**
 * jtest case for react component MenuList
 */
describe('MenuList', () => {
  const menuList = [
    {"displayName": "Main",
     "commands": [
       {"action":"#",
        "displayName":"Home",
        "icon": "fa-th-large"},
        {"action":"#",
        "displayName":"About",
        "icon": "fa-address-card"},
        {"action":"#",
        "displayName":"Services",
        "icon": "fa-cubes"},
        {"action":"#",
        "displayName":"Gallery",
        "icon": "fa-picture-o"}
     ]},
    {"displayName": "Charts",
  "commands": [
    {"action":"#",
        "displayName":"Bar Chart",
        "icon": "fa-bar-chart"},
        {"action":"#",
        "displayName":"Pie Chart",
        "icon": "fa-pie-chart"}
  ]}
  ]

  it('renders correctly', () => {
    const tree = renderer
      .create(<MenuList menuList={menuList} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/**
 * jtest case for react component MenuList
 */
describe('Sidebar', () => {
  const profile = {
    "name": "Nuthan Kumar",
    "designation": "Developer",
    "avatar": "https://test.com"
  }

  const menuList = [
    {"displayName": "Main",
     "commands": [
       {"action":"#",
        "displayName":"Home",
        "icon": "fa-th-large"},
        {"action":"#",
        "displayName":"About",
        "icon": "fa-address-card"},
        {"action":"#",
        "displayName":"Services",
        "icon": "fa-cubes"},
        {"action":"#",
        "displayName":"Gallery",
        "icon": "fa-picture-o"}
     ]},
    {"displayName": "Charts",
  "commands": [
    {"action":"#",
        "displayName":"Bar Chart",
        "icon": "fa-bar-chart"},
        {"action":"#",
        "displayName":"Pie Chart",
        "icon": "fa-pie-chart"}
  ]}
  ]

  it('renders correctly', () => {
    const tree = renderer
      .create(<Sidebar props = { {"profile": profile, "menuList": menuList }}></Sidebar>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
