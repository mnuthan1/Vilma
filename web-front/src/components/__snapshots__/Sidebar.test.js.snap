// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Command renders correctly 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link text-dark font-italic bg-light"
    href="#"
  >
    <i
      className="fa mr-3 text-primary fa-fw fa-th-large"
    />
    Home
  </a>
</li>
`;

exports[`Menu renders correctly 1`] = `
Array [
  <p
    className="text-gray font-weight-bold text-uppercase px-3 small pb-4 mb-0"
  >
    Main
  </p>,
  <ul
    className="nav flex-column bg-white mb-0"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-th-large"
        />
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-address-card"
        />
        About
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-cubes"
        />
        Services
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-picture-o"
        />
        Gallery
      </a>
    </li>
  </ul>,
]
`;

exports[`MenuList renders correctly 1`] = `
Array [
  <p
    className="text-gray font-weight-bold text-uppercase px-3 small pb-4 mb-0"
  >
    Main
  </p>,
  <ul
    className="nav flex-column bg-white mb-0"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-th-large"
        />
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-address-card"
        />
        About
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-cubes"
        />
        Services
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-picture-o"
        />
        Gallery
      </a>
    </li>
  </ul>,
  <p
    className="text-gray font-weight-bold text-uppercase px-3 small pb-4 mb-0"
  >
    Charts
  </p>,
  <ul
    className="nav flex-column bg-white mb-0"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-bar-chart"
        />
        Bar Chart
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-pie-chart"
        />
        Pie Chart
      </a>
    </li>
  </ul>,
]
`;

exports[`Sidebar renders correctly 1`] = `
<div
  className="vertical-nav bg-white"
  id="sidebar"
>
  <div
    className="py-3 px-3 mb-3 bg-light"
  >
    <div
      className="media d-flex align-items-center"
    >
      <img
        alt="..."
        className="mr-3 rounded-circle img-thumbnail shadow-sm"
        src="https://test.com"
        width="65"
      />
      <div
        className="media-body"
      >
        <h4
          className="m-0"
        >
          Nuthan Kumar
        </h4>
        <p
          className="font-weight-light text-muted mb-0"
        >
          Developer
        </p>
      </div>
    </div>
    <div
      className="w-100 row .text-gray"
    >
      <i
        className="fa mr-3 fa-fw fa-power-off"
      />
    </div>
  </div>
  <p
    className="text-gray font-weight-bold text-uppercase px-3 small pb-4 mb-0"
  >
    Main
  </p>
  <ul
    className="nav flex-column bg-white mb-0"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-th-large"
        />
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-address-card"
        />
        About
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-cubes"
        />
        Services
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-picture-o"
        />
        Gallery
      </a>
    </li>
  </ul>
  <p
    className="text-gray font-weight-bold text-uppercase px-3 small pb-4 mb-0"
  >
    Charts
  </p>
  <ul
    className="nav flex-column bg-white mb-0"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-bar-chart"
        />
        Bar Chart
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link text-dark font-italic bg-light"
        href="#"
      >
        <i
          className="fa mr-3 text-primary fa-fw fa-pie-chart"
        />
        Pie Chart
      </a>
    </li>
  </ul>
</div>
`;
